syntax = "proto3";

package posts;

option go_package = "/posts";

service PostService {
 rpc CreatePost (CreatePostRequest) returns (Post) {}
 rpc UpdatePost (UpdatePostRequest) returns (Post) {}
 rpc DeletePost (DeletePostRequest) returns (DeletePostResponse) {}
 rpc GetPostById (GetPostByIdRequest) returns (Post) {}
 rpc ListPosts (ListPostsRequest) returns (ListPostsResponse) {}
}

message Post {
 string id = 1;
 string title = 2;
 string content = 3;
 int64 user_id = 4;
 int64 created_at = 5;
 int64 updated_at = 6;
}

message CreatePostRequest {
 string title = 1;
 string content = 2;
 int64 user_id = 3;
}

message UpdatePostRequest {
 string id = 1;
 string title = 2;
 string content = 3;
 int64 user_id = 4;
}

message DeletePostRequest {
 string id = 1;
 int64 user_id = 2;
}

message DeletePostResponse {
 bool success = 1;
}

message GetPostByIdRequest {
 string id = 1;
}

message ListPostsRequest {
 int64 user_id = 1;
 int32 page = 2;
 int32 limit = 3;
}

message ListPostsResponse {
 repeated Post posts = 1;
 int64 total_count = 2;
}
